torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2085,6 +2085,42 @@ def __init__(self, var_to_sources: Mapping[sympy.Symbol, List[Source]]):
         super().__init__(var_to_sources, lambda n: n.name(), var_to_sources)
 
 
+class ShapeGuardCppPrinter:
+    def __init__(self) -> None:
+        self.printer = CppPrinter()
+        self.all_symbols: Set[str] = set()
+        self.source_to_symbol: Dict[Source, sympy.Symbol] = {}
+
+    def doprint(
+        self, expr: sympy.Expr, symbol_to_sources: Dict[sympy.Symbol, List[Source]]
+    ) -> str:
+        replacements = {}
+        for symbol, sources in symbol_to_sources.items():
+            existing_symbol = self.source_to_symbol.get(sources[0], None)
+            if existing_symbol is None:
+                mangled_name = re.sub("[^0-9a-zA-Z_]+", "_", symbol.name)
+                if symbol.name == mangled_name:
+                    self.source_to_symbol[sources[0]] = symbol
+                else:
+                    old_mangled_name = mangled_name
+                    count = 0
+                    while mangled_name in self.all_symbols:
+                        mangled_name = f"{old_mangled_name}_{count}"
+                        count += 1
```

Comment:
fwiw, we noticed that in FX, if you actually do expect there to be a lot of conflicts, counting up one by one is quite slow. I guess here you don't expect there to be conflicts though...

Reply:
There were a few conflicts like `L['foo'][0]` and `L['foo_0']` both mangling to `L_foo_0`